test/e2e/testdata/complex-global-artifact-passing.yaml

```diff
@@ -155,15 +155,15 @@ spec:
 
 
   - container:
+      image: argoproj/argosay:v2
       command: [sh, -c]
-      args: ["sleep 1; cat /upload/testUpload; cat /upload/testUpload.txt > /upload/testUpload.txt"]
-      image: alpine:3.18.4
+      args: ["sleep 1; cat /upload/testInput.txt > /upload/testUpload.txt"]
```

Comment:
This one was a little tricky, because the behavior of `sh` in alpine is actually different than `sh` in `argoproj/argosay:v2`, because the former allows `cat /upload/testUpload.txt > /upload/testUpload.txt` and the latter does not:
```
 docker run --entrypoint=/bin/sh --rm alpine:3.18.4 -xc 'touch /tmp/foo; cat /tmp/foo > /tmp/foo'; echo $?                                               
+ touch /tmp/foo
+ cat /tmp/foo
0

$ docker run --entrypoint=/bin/sh --rm argoproj/argosay:v2 -xc 'touch /tmp/foo; cat /tmp/foo > /tmp/foo'; echo $?                                         
+ touch /tmp/foo
+ cat /tmp/foo
cat: /tmp/foo: input file is output file
1
```

I think the intention of this was to copy the input artifact to the output, so I updated it to do so